typing/typedecl_separability.ml

```diff
@@ -147,6 +147,7 @@ let rec immediate_subtypes : type_expr -> type_expr list = fun ty ->
       (* these should only occur under Tobject and not at the toplevel,
          but "better safe than sorry" *)
       immediate_subtypes_object_row [] ty
+  | Tquote _ | Tsplice _ -> []
```

Comment:
This case seems less important, but I think `Tquote x` can return `[x]`. Similar for `Tsplice`.